modules/va_forms/app/sidekiq/va_forms/form_builder.rb

```diff
@@ -8,91 +8,124 @@ class FormBuilder
-        # If PDF is not valid, only notify if it was previously valid
+    # returned by the form URL, sends appropriate Slack notifications
+    def send_slack_notifications(existing_form, updated_attrs, content_type)
+      if existing_form.url != updated_attrs[:url]
+        # If the URL has changed, we notify regardless of content
         notify_slack(
-          "URL for form #{updated_attrs[:form_name]} no longer returns a valid PDF or web page.",
-          form_url: updated_attrs[:url]
+          "Form #{updated_attrs[:form_name]} has been updated.",
+          from_form_url: existing_form.url,
+          to_form_url: updated_attrs[:url]
         )
+      elsif existing_form.sha256 != updated_attrs[:sha256] && content_type == 'application/pdf'
+        # If sha256 has changed, only notify if the URL actually returns a PDF, because if the URL is for a download
+        # web page instead, the change in sha256 may not actually reflect a change in the PDF itself
+        notify_slack("PDF contents of form #{updated_attrs[:form_name]} have been updated.")
       end
     end
 
-    # Finds or creates a matching +VAForms::Form+ based on the given +form_data+ as returned from the graphql query
-    def find_or_create_form(form_data)
-      VAForms::Form.find_or_initialize_by row_id: form_data['fieldVaFormRowId']
+    # Finds or initializes a matching +VAForms::Form+ based on the given +form_data+ as returned from the graphql query
+    def find_or_initialize_form(form_data)
+      VAForms::Form.find_or_initialize_by(row_id: form_data['fieldVaFormRowId'])
     end
 
     # Returns a hash of attributes for a +VAForms::Form+ record based on the given +form_data+
     # rubocop:disable Metrics/MethodLength
     def gather_form_attributes(form_data)
-      form_url = normalize_form_url(form_data.dig('fieldVaFormUrl', 'uri'))
-
       attrs = {
         form_name: form_data['fieldVaFormNumber'],
         title: form_data['fieldVaFormName'],
         pages: form_data['fieldVaFormNumPages'],
         language: form_data['fieldVaFormLanguage'].presence || 'en',
         form_type: form_data['fieldVaFormType'],
         form_usage: form_data.dig('fieldVaFormUsage', 'processed'),
-        form_details_url: form_data['entityPublished'] ? "#{FORM_BASE_URL}#{form_data.dig('entityUrl', 'path')}" : '',
+        form_details_url:
+          form_data['entityPublished'] ? "#{VAForms::Form::FORM_BASE_URL}#{form_data.dig('entityUrl', 'path')}" : '',
         form_tool_intro: form_data['fieldVaFormToolIntro'],
         form_tool_url: form_data.dig('fieldVaFormToolUrl', 'uri'),
         deleted_at: form_data.dig('fieldVaFormDeletedDate', 'value'),
         related_forms: form_data['fieldVaFormRelatedForms'].map { |f| f.dig('entity', 'fieldVaFormNumber') },
         benefit_categories: parse_benefit_categories(form_data),
         va_form_administration: form_data.dig('fieldVaFormAdministration', 'entity', 'entityLabel'),
-        url: form_url,
```

Comment:
We are now setting the `url` later in the process (see `check_form_validity` method) to avoid updating it until we have either 1) verified that it returns a successful response or 2) exhausted the number of retries for fetching the URL (7 retries).